### DISCUSSION TITLE
Production Build Error: Web3auth + Vite + Qwik

### DISCUSSION THREAD

**Discussion by reemardelarosa** [Web3Auth/Web3Auth#934] [Help]
https://github.com/Web3Auth/web3auth-web/issues/933

### Issue

Buffer is not defined when deployed to production but working in local development.
Applied most of the solutions here but had no luck.
https://github.com/vitejs/vite/discussions/2785

1. Go to https://wallet.dloyal.com/
2. Open the devtools
3. See the Buffer is not defined error.
4. Signin page is not loading and the website is not usable.
![image](https://user-images.githubusercontent.com/4918140/203891098-5ba72fba-b825-40b6-aafa-be3a11884cd6.png)

Replicated also in stackblitz setup
https://stackblitz.com/edit/qwik-starter-dtpczo?file=adaptors/express/vite.config.ts
1. Wait for the webcontainer to load and dependencies to be installed.
2. run `yarn serve`
3. Open console and see the Buffer error
<img width="1476" alt="Screenshot 2022-11-25 at 11 09 02 AM" src="https://user-images.githubusercontent.com/4918140/203892678-c3430aee-f17c-4fef-a531-746a563a4d3e.png">

**In the local build, it is working.**
<img width="1379" alt="Screenshot 2022-11-25 at 11 13 12 AM" src="https://user-images.githubusercontent.com/4918140/203893156-3f9c1716-7d54-4b32-8747-0ecf9fff90b8.png">

https://user-images.githubusercontent.com/4918140/203891353-35f17394-33b5-4db1-a209-d6225701816d.mov
### **Device Info (please complete the following information):**

- Device: Mac Book
- OS: IOS
- Browser Chrome
- Version 107.0.5304.110 (Official Build) (arm64)
- Web3Auth Version
```json
 "@web3auth/base": "3.0.4",
 "@web3auth/core": "3.0.4",
 "@web3auth/openlogin-adapter": "3.0.4",
```

### **Additional context**
For reference 
- https://github.com/ionic-team/rollup-plugin-node-polyfills

vite.config.ts
```js
import nodePolyfills from 'rollup-plugin-node-polyfills';
import { expressAdaptor } from '@builder.io/qwik-city/adaptors/express/vite';
import { extendConfig } from '@builder.io/qwik-city/vite';
import baseConfig from '../../vite.config';

export default extendConfig(baseConfig, () => {
  return {
    build: {
      ssr: true,
      rollupOptions: {
        input: ['src/entry.express.tsx', '@qwik-city-plan'],
        plugins: [
          // Enable rollup polyfills plugin
          // used during production bundling
          nodePolyfills()
        ]
      },
    },
    plugins: [
      expressAdaptor({
        staticGenerate: true,
      }),
    ],
  };
});

```
Tried from related discussion https://github.com/orgs/Web3Auth/discussions/729
```js
export default extendConfig(baseConfig, () => {
  return {
    build: {
      ssr: true,
      target: 'es2020',
      commonjsOptions: {
        transformMixedEsModules: true
      },
      rollupOptions: {
        input: ['src/entry.express.tsx', '@qwik-city-plan'],
      },
    },
    plugins: [
      expressAdaptor({
        staticGenerate: true,
      }),
    ],
  };
});
```

**Comment by reemardelarosa** [comment]
Tried similar solution here but got no luck.
https://github.com/orgs/Web3Auth/discussions/729

**Comment by chaitanyapotti** [comment]
This is fixed in 4.4.2
